src/video_core/renderer_vulkan/vk_staging_buffer_pool.cpp

```diff
@@ -61,14 +61,16 @@ std::optional<u32> FindMemoryTypeIndex(const VkPhysicalDeviceMemoryProperties& p
     return std::nullopt;
 }
 
-u32 FindMemoryTypeIndex(const VkPhysicalDeviceMemoryProperties& props, u32 type_mask) {
-    // Try to find a DEVICE_LOCAL_BIT type, Nvidia and AMD have a dedicated heap for this
-    std::optional<u32> type = FindMemoryTypeIndex(props, type_mask, STREAM_FLAGS);
-    if (type) {
-        return *type;
+u32 FindMemoryTypeIndex(const VkPhysicalDeviceMemoryProperties& props, u32 type_mask, bool device_local) {
+    if (device_local) {
```

Comment:
```suggestion
u32 FindMemoryTypeIndex(const VkPhysicalDeviceMemoryProperties& props, u32 type_mask, bool try_device_local) {
    if (try_device_local) {
```

Since it doesn't guarantee that it will be device local.